README.md

```diff
@@ -3,7 +3,7 @@
 ![GitHub](https://img.shields.io/github/license/mbg/wai-saml2)
 ![Haskell CI](https://github.com/mbg/wai-saml2/workflows/Haskell/badge.svg?branch=master)
 
-A Haskell library which implements SAML2 assertion validation as WAI middleware. This can be used by a Haskell web application (the service provider, SP) to perform identity provider (IdP) initiated authentication, i.e. SAML2-based authentication where the user is redirected to the IdP by the SP, the IdP authenticates the user, and then gets the user to submit a SAML2 assertion back to the SP. 
+A Haskell library which implements SAML2 assertion validation as WAI middleware. This can be used by a Haskell web application (the service provider, SP) to perform identity provider (IdP) initiated authentication, i.e. SAML2-based authentication where the authentication begins at the IdP-end, the IdP authenticates the user, and then gets the user to submit a SAML2 assertion back to the SP (known as "unsolicited SSO" within e.g. [the Shibboleth project](https://wiki.shibboleth.net/confluence/display/IDP30/UnsolicitedSSOConfiguration#UnsolicitedSSOConfiguration-SAML2.0)). 
```

Comment:
I've re-worded this, because having the SP redirect the user to the IdP-initiated auth endpoint is unusual and probably not the normal circumstances you'd see IdP-initiated authentication used in.

E.g. for AD FS, users can login to the IdP itself and choose from a dropdown of service providers to initiate unsolicited auth that way (without hitting the service first).